cypress/integration/simple_spec.ts

```diff
@@ -1,135 +1,141 @@
-      .should('match', /1-\d+ of \d+ results for/);
+    const searchDetailsPage = new SearchDetailsPo();
+    const notEmptyResultsRegex = /1-\d+ of \d+ results for/;
+    const defaultSortOption = 'featured-rank';
+    const departmentName = 'Computers';
+    const productForSearch = 'mouse';
+
+    homePage.clickOnElem(homePage.shopByCategoryLink);
+    homePage.clickOnElem(homePage.computerDepLink);
+
+    searchDetailsPage.isElemContainText(searchDetailsPage.departmentTitle, departmentName);
+    searchDetailsPage.isElemTextMatch(searchDetailsPage.resultsCounter, notEmptyResultsRegex);
+    searchDetailsPage.isElemHaveValue(searchDetailsPage.sortDropdown, defaultSortOption);
+
+    searchDetailsPage.typeIntoInput(searchDetailsPage.searchInput, productForSearch);
+    searchDetailsPage.clickOnElem(searchDetailsPage.searchIcon);
+
+    searchDetailsPage.isElemContainText(searchDetailsPage.afterSearchBreadcumb, departmentName);
+    searchDetailsPage.isElemContainText(searchDetailsPage.afterSearchBreadcumb, productForSearch);
+    searchDetailsPage.isElemTextMatch(searchDetailsPage.afterSearchResCounter, notEmptyResultsRegex);
   });
 
   it('Product filtering by Deal type check', () => {
-    cy.get('.nav-a').contains('Today\'s Deals').click();
+    const dealsPage = new DealsPo();
+    const defaulSortOption = 'Relevance';
+    const dealsTitleText = 'Deals';
+
+    homePage.clickOnElemWithText(homePage.headerNavLinks, homePage.dealLinkText);
 
-    cy.get('.gbh1-bold')
-      .should('to.contain', 'Deals')
-      .and('to.be.visible');
-    cy.get('.a-dropdown-prompt')
-      .should('to.contain', 'Relevance');
+    dealsPage.isElemContainText(dealsPage.pageTitle, dealsTitleText);
+    dealsPage.isElemContainText(dealsPage.sortDropdown, defaulSortOption);
 
-    cy.get('[data-value="DEAL_OF_THE_DAY"]').click();
+    dealsPage.clickOnElem(dealsPage.dealOfTheDayFilter);
 
-    cy.get('.dealTile')
-      .each($productTile => cy.wrap($productTile)
-        .should('to.have.descendants', '.dotdBadge'));
+    dealsPage.isAllProdLabelled(dealsPage.productCard, dealsPage.dealOfDayLabel);
   });
 
-  // should be reworked due to several types of logic for the product adding
-  it.skip('Product ordering functionality check', () => {
-    cy.get('#twotabsearchtextbox').type('laptop');
-    cy.get('.nav-search-submit').click();
+  it('Product ordering functionality check', () => {
+    const searchDetailsPage = new SearchDetailsPo();
+    const productDetailsPage = new ProductDetailsPo();
+    const cartPage = new CartPo();
+    const productToSearch = 'headphones';
+    const notEmptyResultsRegex = /1-\d+ /;
+    const alertText = 'Added to Cart';
+    const cartTotal = '1 item';
+    const signInUrl = 'https://www.amazon.com/ap/signin';
+
+    homePage.typeIntoInput(homePage.searchInput, productToSearch);
+    homePage.clickOnElem(homePage.searchIcon);
+
+    homePage.isElemTextMatch(searchDetailsPage.afterSearchResCounter, notEmptyResultsRegex);
```

Comment:
looks like it's already searchDetailsPage, so use searchDetailsPage.isElemTextMatch